src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -2411,14 +2412,23 @@ private static bool CanConvert(Expression expression, DocumentConventions conven
 
         public static ParameterExpression GetParameter(MemberExpression expression)
         {
-            return GetInnermostExpression(expression, out _) as ParameterExpression;
+            return GetInnermostExpression(expression, out _, out _) as ParameterExpression;
         }
 
-        public static Expression GetInnermostExpression(MemberExpression expression, out string path)
+        private static ParameterExpression GetParameterAndCheckInternalMemberName(MemberExpression expression, out bool hasInternal)
+        {
+            return GetInnermostExpression(expression, out _, out hasInternal) as ParameterExpression;
+        }
+
+        public static Expression GetInnermostExpression(MemberExpression expression, out string path, out bool hasInternal)
         {
             path = string.Empty;
+            hasInternal = false;
```

Comment:
```suggestion
            hasInternalKeyOrValue = false;
```

Reply:
Caution, hasInternal variable is used too in line 265

Reply:
done